### PR TITLE
add a tooltip to the encoding status view

### PR DESCRIPTION
**Pull request by bronson**
### Description of the Change

This PR adds a tooltip to the status bar indicator.

### Alternate Designs

No other designs were considered.  Happy to make changes.

### Benefits

Unfamiliar users will be reassured that, yes, this indicator does refer to the file being edited.

Also, most status bar items have tooltips.  This offers a bit of reassuring consistency.

### Possible Drawbacks

Maybe I'll leak the tooltip if I missed a place where it needs to be disposed?

### Applicable Issues

#26

### CODE REVIEW

**Comment by bronson** [comment]
Right now the wording is: `File displayed using UTF-8 encoding`.

File: lib/encoding-status-view.js [hunk 1]
```diff
@@ -64,6 +68,11 @@ export default class EncodingStatusView {
         this.encodingLink.textContent = encodingLabel.status
         this.encodingLink.dataset.encoding = editorEncoding
         this.element.style.display = ''
+
+        if(this.toolTip) {
+          this.toolTip.dispose()
+        }
+        this.toolTip = atom.tooltips.add(this.encodingLink, {title: 'File displayed using ' + encodingLabel.list + ' encoding'})
```
Comment (50Wliu) [inline lib/encoding-status-view.js]: I think we generally use `this.tooltip`.

Comment (bronson) [inline lib/encoding-status-view.js]: Yes, that's better.  Changed.

File: lib/encoding-status-view.js [hunk 2]
```diff
@@ -64,6 +68,11 @@ export default class EncodingStatusView {
         this.encodingLink.textContent = encodingLabel.status
         this.encodingLink.dataset.encoding = editorEncoding
         this.element.style.display = ''
+
+        if(this.tooltip) {
```
Comment (damieng) [inline lib/encoding-status-view.js]: Need a space between if and (

File: lib/encoding-status-view.js [hunk 3]
```diff
@@ -64,6 +68,11 @@ export default class EncodingStatusView {
...
+          this.tooltip.dispose()
+        }
+        this.tooltip = atom.tooltips.add(this.encodingLink, {title: 'File displayed using ' + encodingLabel.list + ' encoding'})
```
Comment (damieng) [inline lib/encoding-status-view.js]: I wonder if 'displayed' is the right word. It's parsed with the encoding so if you change encoding and save it you can lose information.  Displayed sounds like it's completely passive.  Also consider using string interpolation.

Comment (bronson) [inline lib/encoding-status-view.js]: Very good point.  Changed to `This file uses UTF-8 encoding`.

I wasn't sure I wanted to be the first to use interpolation in this package... but it works fine.  :)

File: lib/encoding-status-view.js (hunk 2 above)
Comment (bronson) [inline lib/encoding-status-view.js]: Fixed.  And installed the linter-js-standard package.

**Pull request merged**